mesonbuild/scripts/meson_install.py

```diff
@@ -29,13 +29,21 @@ def append_to_log(line):
     install_log_file.flush()
 
 def do_copy(from_file, to_file):
-    try:
-        # Python's copyfile fails if the target file already exists.
-        os.unlink(to_file)
-    except FileNotFoundError:
-        pass
-    shutil.copyfile(from_file, to_file)
-    shutil.copystat(from_file, to_file)
+    if os.path.isfile(from_file):
+        # copyfile fails if the target file already exists, so remove it to
+        # allow overwriting a previous install. If the target is not a file, we
+        # don't try to remove it since it's probably a user error.
+        if os.path.isfile(to_file):
+            os.unlink(to_file)
+        shutil.copyfile(from_file, to_file)
+        shutil.copystat(from_file, to_file)
+    else:
+        # copytree fails if the target dir already exists, so remove it to
+        # allow overwriting a previous install. If the target is not a dir, we
+        # don't try to remove it since it's probably a user error.
+        if os.path.isdir(to_file):
+            shutil.rmtree(to_file, ignore_errors=True)
```

Comment:
This is very dangerous. It will delete all files in the target dir, even those put in by someone else. This can nuke a lot of stuff, especially if the target dir is `/usr` or something similar. It would be better if this used `install_subdirs` instead (it might need refactoring, though). `do_copy` is perhaps a bit poorly named as its job is only to copy files. Maybe it should be renamed to `do_copy_file`.